src/pages/settings/Wallet/ActivateCardPage.js

```diff
@@ -0,0 +1,175 @@
+import React, {useRef, useCallback, useState, useEffect} from 'react';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import {withOnyx} from 'react-native-onyx';
+import lodashGet from 'lodash/get';
+import {isUndefined} from 'lodash';
+import Text from '../../../components/Text';
+import Navigation from '../../../libs/Navigation/Navigation';
+import styles from '../../../styles/styles';
+import MagicCodeInput from '../../../components/MagicCodeInput';
+import * as DeviceCapabilities from '../../../libs/DeviceCapabilities';
+import * as ErrorUtils from '../../../libs/ErrorUtils';
+import * as CardSettings from '../../../libs/actions/CardSettings';
+import BigNumberPad from '../../../components/BigNumberPad';
+import Button from '../../../components/Button';
+import IllustratedHeaderPageLayout from '../../../components/IllustratedHeaderPageLayout';
+import themeColors from '../../../styles/themes/default';
+import SCREENS from '../../../SCREENS';
+import * as LottieAnimations from '../../../components/LottieAnimations';
+import useWindowDimensions from '../../../hooks/useWindowDimensions';
+import ONYXKEYS from '../../../ONYXKEYS';
+import useLocalize from '../../../hooks/useLocalize';
+import ROUTES from '../../../ROUTES';
+import assignedCardPropTypes from './assignedCardPropTypes';
+
+const propTypes = {
+    /* Onyx Props */
+    /** The details about the Expensify cards */
+    cardList: PropTypes.shape({
+        isLoading: PropTypes.bool,
+        [PropTypes.string]: PropTypes.objectOf(assignedCardPropTypes),
+        [PropTypes.string]: PropTypes.shape({
+            // eslint-disable-next-line react/forbid-prop-types
+            errors: PropTypes.object,
+        }),
```

Comment:
1. Can we expect when there is a shape of cards, we always have `isLoading`? 
2. Can we extend a card object?

```suggestion
        isLoading: PropTypes.bool.isRequired,
        [PropTypes.string]: PropTypes.objectOf({
            ...assignedCardPropTypes,   
            errors: PropTypes.object.isRequired,
        }),
```

Reply:
![image](https://github.com/pasyukevich/Expensify-app/assets/23487612/4eea8180-bb85-4e2c-83c9-649fb0ef8f0d)

This is what ill be expecting in onyx. Errors are going to be said on the object under the cardID key. Not sure if there is a nicer way to shape props in such scenario.

As for isLoading, I agree

Reply:
Ah, I see. Therefore, no sense to extend a card object with errors.

Reply:
I had to re-do the props completely because it turns out that the structure I was expecting was incorrect. Also after a second thought, isLoading cant be required prop. Its not gonna be there in the begining, its added after 1st API call

Reply:
Thank you for letting me know 🙂